core-plugins/src/main/java/co/cask/hydrator/plugin/common/KinesisUtil.java

```diff
@@ -0,0 +1,159 @@
+ * Utility methods to help create and manage Kinesis Streams
+ */
+public final class KinesisUtil {
+
+  private static final Logger LOG = LoggerFactory.getLogger(KinesisUtil.class);
+  private static final long DELETE_TIMEOUT = 1000 * 120;
+  private static final long CREATE_TIMEOUT = 3 * 60 * 1000;
+  private static final long SLEEP_INTERVAL = 1000 * 10;
+
+  private static final String STATUS_CREATING = "CREATING";
+  private static final String STATUS_DELETING = "DELETING";
+  private static final String STATUS_ACTIVE = "ACTIVE";
+  private static final String STATUS_UPDATING = "UPDATING";
+
+
+  private KinesisUtil() {
+  }
+
+  /**
+   * Creates an Amazon Kinesis stream if it does not exist and waits for it to become available
+   *
+   * @param kinesisClient The {@link AmazonKinesisClient} with Amazon Kinesis read and write privileges
+   * @param streamName The Amazon Kinesis stream name to create
+   * @param shardCount The shard count to create the stream with
+   * @throws IllegalStateException Invalid Amazon Kinesis stream state
+   * @throws IllegalStateException Stream does not become active before the timeout
+   */
+  public static void createAndWaitForStream(AmazonKinesisClient kinesisClient,
+                                            String streamName,
+                                            int shardCount) {
+    if (streamExists(kinesisClient, streamName)) {
+      String state = getStreamState(kinesisClient, streamName);
+      switch (state) {
+        case STATUS_DELETING:
+          long waitTimeDelete = System.currentTimeMillis() + DELETE_TIMEOUT;
+          while (System.currentTimeMillis() < waitTimeDelete && streamExists(kinesisClient, streamName)) {
+            try {
+              LOG.debug("...Deleting Stream {} ...", streamName);
+              Thread.sleep(SLEEP_INTERVAL);
+            } catch (InterruptedException e) {
+              Thread.currentThread().interrupt();
+            }
+          }
+          if (streamExists(kinesisClient, streamName)) {
+            throw new IllegalStateException(String.format("KinesisUtils timed out waiting for stream {} to delete",
+                                                          streamName));
+          }
+          createStream(streamName, shardCount, kinesisClient);
+          break;
+        case STATUS_ACTIVE:
+          LOG.info("Stream {} is ACTIVE", streamName);
+          return;
+        case STATUS_CREATING:
+          LOG.info("Stream {} is being created", streamName);
+          break;
+        case STATUS_UPDATING:
+          LOG.info("Stream {} is UPDATING", streamName);
+          return;
+        default:
+          throw new IllegalStateException("Illegal stream state: " + (state.equals("") ? "State not found" : state));
+      }
+    } else {
+      createStream(streamName, shardCount, kinesisClient);
+    }
+    long waitTimeCreate = System.currentTimeMillis() + CREATE_TIMEOUT;
+    while (System.currentTimeMillis() < waitTimeCreate) {
```

Comment:
not sure what you mean. Both of them should be a streamExists check right? For delete, you bail if stream exists after timeout, and for create, you bail if stream does not exist after timeout?